pkg/machinery/config/configpatcher/configpatcher.go

```diff
@@ -0,0 +1,33 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package configpatcher provides methods to patch Talos config.
+package configpatcher
+
+import (
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	ghodssyaml "github.com/ghodss/yaml"
+)
+
+// JSON6902 is responsible for applying a JSON 6902 patch to the bootstrap data.
+func JSON6902(decodedData []byte, patch jsonpatch.Patch) ([]byte, error) {
```

Comment:
Seems like we could do an easy test on this. Preferably table style (`Go: Generate Unit Tests For Function`) so that we can get a lot of variation on input data.